cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ExecutionSparkContext.java

```diff
@@ -0,0 +1,376 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.twill.api.RunId;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link SparkContext} implementation that is used in both {@link SparkProgram#run(SparkContext)} and
+ * inside Spark executor.
+ */
+public class ExecutionSparkContext extends AbstractSparkContext {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExecutionSparkContext.class);
+
+  // Cache of Dataset instances that are created through this context
+  // for closing all datasets when closing this context. This list does not includes datasets
+  // created through the readFromDataset and writeToDataset methods.
+  private final Map<String, Dataset> datasets;
+  private final Configuration hConf;
+  private final Transaction transaction;
+  private final StreamAdmin streamAdmin;
+  private final DatasetInstantiator datasetInstantiator;
+  private boolean stopped;
+  private ApacheSparkContext apacheSparkContext;
+
+  public ExecutionSparkContext(SparkSpecification specification, Id.Program programId, RunId runId,
+                               ClassLoader programClassLoader, long logicalStartTime,
+                               Map<String, String> runtimeArguments,
+                               Transaction transaction, DatasetFramework datasetFramework,
+                               DiscoveryServiceClient discoveryServiceClient,
+                               MetricsCollectionService metricsCollectionService,
+                               Configuration hConf, StreamAdmin streamAdmin) {
+    this(specification, programId, runId, programClassLoader, logicalStartTime, runtimeArguments,
+         transaction, datasetFramework, discoveryServiceClient,
+         createMetricsContext(metricsCollectionService, programId, runId),
+         createLoggingContext(programId, runId), hConf, streamAdmin);
+  }
+
+  public ExecutionSparkContext(SparkSpecification specification, Id.Program programId, RunId runId,
+                                ClassLoader programClassLoader, long logicalStartTime,
+                                Map<String, String> runtimeArguments,
+                                Transaction transaction, DatasetFramework datasetFramework,
+                                DiscoveryServiceClient discoveryServiceClient, MetricsContext metricsContext,
+                                LoggingContext loggingContext, Configuration hConf, StreamAdmin streamAdmin) {
+    super(specification, programId, runId, programClassLoader, logicalStartTime,
+          runtimeArguments, discoveryServiceClient, metricsContext, loggingContext);
+
+    this.datasets = new HashMap<>();
```

Comment:
It's longer, so no. Also the new Java 7 syntax is meant to replace guava usage in this case.